test/Microsoft.AspNetCore.Hosting.Tests/WebHostTests.cs

```diff
@@ -121,6 +123,31 @@ public void UsesNewConfigurationOverLegacyConfigForAddresses()
         }
 
         [Fact]
+        public void DoNotPreferHostingUrlsWhenNoAddressConfigured()
+        {
+            var host = CreateBuilder().UseServer(this).PreferHostingUrls().Build();
+            host.Start();
```

Comment:
Probably, though we never had a problem with it since these are using a fake server, it doesn't consume any ips/ports. The hosts will be GC'ed and disposed automatically.